peps/pep-9999.rst

```diff
@@ -0,0 +1,233 @@
+
+
+Objections
+==================
+
+There are only a few hard objections to the introduction of this syntactic sugar. Most of those not in favour of this feature are simply in the camp of 'I wouldn't use it'. However over the extensive conversations about this feature, the following objections were the most common:
+
+The syntax is ugly
+--------------------------------------------
+This objection is by far the most common. On the contrary, we argue that:
+
+* This objection is is subjective and many community members disagree
+* This is already established syntax used in f-strings
+* Programmers will, as ever, adjust over time
+
+The feature is confusing
+--------------------------------------------
+We argue that:
+
+* Introducing new features typically has this temporary impact
+* The syntax is very similar to the established ``f'{x=}'`` syntax
+* The feature and syntax are familiar from other popular modern languages
+*  The expansion of ``x=`` to ``x=x`` is in fact a trivial feature and inherently significantly less complex than ``*arg`` and ``**kwarg`` expansion
+* This particular syntactic form has been independently proposed on numerous occasions, indicating that it is the most obvious [8]_ [9]_ [13]_
+
+The feature is not explicit
+--------------------------------------------
+This is based on a misunderstanding of the Zen of Python. Keyword arguments are fundamentally more explicit than positional ones where argument assignment is only visible at the function definition. On the contrary, the proposed syntactic sugar contains all the information as is conveyed by the established keyword argument syntax but without the redundancy. Moreover, the introduction of this syntactic sugar incentivises use of keyword arguments, making practical Python code more explicit.
+
+The feature adds another way of doing things
+--------------------------------------------
+
+The same argument can be made against all syntax changes. This is a simple syntactic sugar, much as ``x += 1`` is sugar for ``x = x + 1``. This isn't tantamount to a 'new way' of passing arguments but a more readable notation for the same way.
+
+Renaming the variable in the calling context will break the code
+--------------------------------------------
+
+An ``NameError`` would make the mistake abundantly clear. Moreover, text editors could highlight this based on static analysis ‒ f(x=) is exactly equivalent to writing ``f(x=x)``. If ``x`` does not exist, editors currently have no problem highlighting that issue.
+
+Recommendations
+========================
+As with any other language feature, the programmer should be judicious about whether using this feature  improves their code based on the context. We do not advocate for enforcing a rule of using this feature in all cases where it may be applicable.
```

Comment:
May want some more concrete recommendations.

Context:
> I would appreciate a few examples, in any PEP for this, which show when it is not appropriate. I think that would be enough to guide people away from overzealously trying to enforce this usage on everyone.

Reply:
> the programmer should be judicious about whether using this feature  improves their code based on the context

Sounds weird to me. How about "the programmer needs to exercise their own judgement..."?

Although I think the entire paragraph is a bit too defensive and maybe not needed?

Reply:
Sorry I didn't follow up on your comment although I did reword the paragraph. Let me know if you still l

>Although I think the entire paragraph is a bit too defensive and maybe not needed?

The context behind adding this was [this section of the discussion](https://discuss.python.org/t/syntactic-sugar-to-encourage-use-of-named-arguments/36217/94?u=joshuabambrick). There were a few who noted that the universal use of this syntax in all applicable contexts was not advisable based on [my linked PR](https://discuss.python.org/t/syntactic-sugar-to-encourage-use-of-named-arguments/36217/72?u=joshuabambrick). I generally agree and thought that adding a comment to indicate that we aren't recommending that the use of this feature be enforced universally would help address that concern. There were suggestions at giving some examples as to where it would be most useful but my sense is that these are best left to the discretion of the programmer.